src/CssToInlineStyles.php

```diff
@@ -143,27 +177,39 @@ public function convert($outputXHTML = false)
         $this->processCSS();
 
         // create new DOMDocument
-        $document = new \DOMDocument('1.0', $this->getEncoding());
+        $document = new \DOMDocument();
 
         // set error level
         $internalErrors = libxml_use_internal_errors(true);
 
         // load HTML
-        $document->loadHTML($this->html);
+        $document->loadHTML('<?xml encoding="' . $this->getEncoding() . '">' . $this->html);
 
-        // Restore error level
+        // remove the "xml-encoding" hack
+        foreach ($document->childNodes as $item) {
+            if ($item->nodeType == XML_PI_NODE) {
+                $document->removeChild($item);
+            }
+        }
+
+        // set encoding
+        $document->encoding = $this->getEncoding();
+
+        // restore error level
         libxml_use_internal_errors($internalErrors);
 
         // create new XPath
         $xPath = new \DOMXPath($document);
 
         // any rules?
-        if (!empty($this->cssRules)) {
+        if (0 !== count($this->cssRules)) {
```

Comment:
I think that ``!empty(`` is more readable than ``0 !== count(`` here. It makes the intent clearer